src/Geometry/Primitives/NonCSG/Hexagon.jl

```diff
@@ -75,6 +74,23 @@ function surfaceintersection(hex::Hexagon{T}, r::AbstractRay{T,3}) where {T<:Rea
     end
 end
 
+"""Returns the vertices of the Hexagon represented in the local coordinate frame. The vertices lie in the z = 0 plane and are 2D"""
+function vertices(hex::Hexagon{T}) where{T<:Real}
```

Comment:
@galran yes. I added one. However it allocates. Tried many different approaches but all of them allocated. Since vertices will be called in the innermost loop of multi-lens simulation this could be a problem for multithreading. Will try again tomorrow.

Reply:
@alfredclwong I believe the other vertices() functions do not allocate. Another alternative is to do this upon creation of the object, since the vertices are defined in the local coordinate frame and never change.